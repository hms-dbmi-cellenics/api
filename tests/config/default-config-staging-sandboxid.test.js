// @ts-nocheck
const _ = require('lodash');

const AWS = require('aws-sdk');
const isPromise = require('../../src/utils/isPromise');

jest.mock('aws-sdk');
jest.mock('../../src/utils/getLogger');

describe('default-config', () => {
  const OLD_ENV = _.cloneDeep(process.env);

  beforeEach(() => {
    jest.clearAllMocks();
    process.env = OLD_ENV;
  });

  afterAll(() => {
    jest.clearAllMocks();
    process.env = OLD_ENV;
  });

  it('Returns correct values for staging sandboxid', () => {
    const stagingEnvironment = 'staging';
    process.env.K8S_ENV = stagingEnvironment;
    process.env.CLUSTER_ENV = stagingEnvironment;
    process.env.SANDBOX_ID = 'mockedSandboxId';
    process.env.AWS_DEFAULT_REGION = 'eu-west-1';
    process.env.AWS_REGION = 'eu-west-1';
    process.env.RDS_SANDBOX_ID = 'default';
    process.env.AWS_ROLE_ARN = 'arn:aws:iam::12341234:role/api-role-production';
    process.env.AWS_STS_REGIONAL_ENDPOINTS = 'regional';
    process.env.AWS_WEB_IDENTITY_TOKEN_FILE = 'AWS_WEB_IDENTITY_TOKEN_FILEMocked';
    process.env.AWS_XRAY_DAEMON_ADDRESS = 'AWS_XRAY_DAEMON_ADDRESSMocked';
    process.env.DOMAIN_NAME = 'scp-staging.biomage.net';

    const userPoolId = 'mockUserPoolId';
    const accountId = 'mockAccountId';

    AWS.CognitoIdentityServiceProvider = jest.fn(() => ({
      listUserPools: {
        promise: jest.fn(() => Promise.resolve(
          { UserPools: [{ id: userPoolId, Name: `biomage-user-pool-case-insensitive-${stagingEnvironment}` }] },
        )),
      },
    }));

    AWS.STS = jest.fn(() => ({
      getCallerIdentity: {
        promise: jest.fn(() => Promise.resolve({ Account: accountId })),
      },
    }));

    const defaultConfig = jest.requireActual('../../src/config/default-config');

    const defaultConfigEntries = Object.entries(defaultConfig);
    const filteredEntries = defaultConfigEntries.filter(([, value]) => !isPromise(value));
    const defaultConfigFiltered = Object.fromEntries(filteredEntries);

    expect(defaultConfigFiltered).toMatchSnapshot();
  });
});
