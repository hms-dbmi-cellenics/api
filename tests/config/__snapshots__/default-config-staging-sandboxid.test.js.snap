// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default-config Returns correct values for staging sandboxid 1`] = `
Object {
  "api": Object {
    "prefix": "/",
  },
  "awsAccountId": "000000000000",
  "awsRegion": "eu-west-1",
  "cachingEnabled": false,
  "clusterEnv": "staging",
  "cognitoISP": Object {
    "listUserPools": Object {
      "promise": [MockFunction],
    },
  },
  "corsOriginUrl": Array [
    "https://sandbox.elabjournal.com",
    "https://elabjournal.com",
    "https://ui-mockedSandboxId.scp-staging.biomage.net",
  ],
  "datadogApiKey": "",
  "datadogAppKey": "",
  "domainName": "scp-staging.biomage.net",
  "emailDomainName": "https://ui-mockedSandboxId.scp-staging.biomage.net",
  "pipelineIgnoreSSLCertificate": false,
  "pipelineInstanceConfigUrl": "https://raw.githubusercontent.com/org/releases/master/staging/mockedSandboxId.yaml",
  "pipelineNamespace": "pipeline-mockedSandboxId",
  "podName": "local",
  "port": 3000,
  "publicApiUrl": "https://api-mockedSandboxId.scp-staging.biomage.net",
  "rdsSandboxId": "default",
  "sandboxId": "mockedSandboxId",
  "workerNamespace": "worker-mockedSandboxId",
  "workerVersion": 4,
}
`;
